src/main/java/org/jvnet/hudson/test/junit/jupiter/JenkinsRuleExtension.java

```diff
@@ -0,0 +1,36 @@
+package org.jvnet.hudson.test.junit.jupiter;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.runner.Description;
+import org.jvnet.hudson.test.JenkinsRecipe;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import javax.annotation.Nonnull;
+import java.lang.reflect.Method;
+import java.util.Optional;
+
+/**
+ * Provides JUnit 5 compatibility for {@link JenkinsRule}.
+ */
```

Comment:
See Jesse's comment below: make clear that the rule does create a new Jenkins instance per test.

Reply:
Added to `@JenkinsRule`.